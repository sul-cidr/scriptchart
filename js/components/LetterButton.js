import React from "react";

/* LetterButton - Creats a <span> that contains a Bulma
 * button item, which contains a <SyriacLetter> component.
 * When clicked, it either signals the DashTabs and
 * ScriptChart component to redisplay the letter's row
 * (if this is a button in the ChartAccordion list)
 * or adds/removes the letter to the list of selected
 * letters (if this is a button in the configuration form's
 * letters grid).
 */

import SyriacLetter from "./SyriacLetter";

class LetterButton extends React.Component {
  constructor(props) {
    super(props);

    this.onLetterClicked = this.onLetterClicked.bind(this);
  }
  onLetterClicked() {
    if (this.props.hasOwnProperty("onHiddenChange")) {
      this.props.onHiddenChange("show", "row", this.props.letterID);
    } else {
      if (this.props.buttonClass === "button is-outlined") {
        this.props.onLetterClick(this.props.letterID, "select");
      } else {
        this.props.onLetterClick(this.props.letterID, "deselect");
      }
    }
  }
  render() {
    return (
      <span className={this.props.buttonClass} onClick={this.onLetterClicked}>
        <SyriacLetter
          id={this.props.letterID}
          clicked={this.props.clicked}
          isButton={this.props.isButton}
        />
      </span>
    );
  }
}

export default LetterButton;
