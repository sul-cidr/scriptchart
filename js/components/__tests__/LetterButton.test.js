/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from "react";
import renderer from "react-test-renderer";
import LetterButton from "../LetterButton";

describe("LetterButton test", () => {
  it("LetterButton should match snapshot", () => {
    const component = renderer.create(<LetterButton />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
