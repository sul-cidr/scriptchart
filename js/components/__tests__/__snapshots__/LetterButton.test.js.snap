// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LetterButton test LetterButton should match snapshot 1`] = `
<span
  onClick={[Function]}
/>
`;
